src/main/java/io/opentracing/contrib/spanmanager/ThreadLocalSpanManager.java

```diff
@@ -0,0 +1,100 @@
+package io.opentracing.contrib.spanmanager;
+
+import io.opentracing.NoopSpan;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default {@link SpanManager} implementation using {@link ThreadLocal} storage,
+ * maintaining a stack-like structure of linked managed spans.
+ * <p>
+ * The linked managed spans provide the following stack unwinding algorithm:
+ * <ol>
+ * <li>If the released <em>span</em> is not the managed span, the managed span is left alone.</li>
+ * <li>Otherwise, the first parent that is <em>not yet released</em> is set as the new managed span.</li>
+ * <li>If no managed parents remain, the <em>managed span</em> is cleared.</li>
+ * <li>Consecutive <code>release()</code> calls for already-released spans will be ignored.</li>
+ * </ol>
+ */
+final class ThreadLocalSpanManager implements SpanManager {
```

Comment:
Good question. The idea is that people can keep using their own context propagation mechanism they may already have, as that has impact on the code.
For instance, I created a security context propagation for a client that transfers to new background threads. I can easily create a SpanManager adaptor using that same propagation tool to _also_ propagate/piggyback the spans besides other contextual information.

The impact is on the code scheduling the new threads which needs to be decoupled from whatever context must be propagated.
Whatever works for somebody should be extendable to also propagate spans with, hence the SpanManager interface.

I agree there needs to be cleanup and clarification. What is now the ActiveSpanManager is merely a 'global' placeholder similar to the 'global' tracer. 
Should it be named GlobalSpanManager again? similar to GlobalTracer??

The `concurrent` package is meant to provide an _example_ for propagating spans accross threads, but by no means is anyone obliged to use it.
I personally prefer an existing context propagation solution to which the SpanManager is registered as callback.
I have this concept working in a library, but it is currently not documented.

The `tracer` package is meant to automatically _activate_ / _deactivate_ spans to be propagated when _started_ / _finished_.
Again, this is not obligatory, but in my opinion a very useful tool to separate the tracing from the propagation.